runtime/lua/vim/fs.lua

```diff
@@ -275,6 +280,24 @@ function M.find(names, opts)
   local stop = opts.stop
   local limit = opts.limit or 1
 
+  if opts.dfs then
+    local matches = {}
+    if type(names) == 'function' then
+      error("Can't use dfs with functions")
+    end
```

Comment:
What's the reason for this restriction?

Reply:
It didn't seem to make sense, because if `names` is a function then `dfs` should not affect it? Although we could still allow it tbf.